### ISSUE TITLE
Add arrays

### ISSUE THREAD

**Issue by effectfully** [effectfully/tiny-lang#29]
We now have loops and statically unrollable arrays will give us a lot of power.

**Comment by kwxm** [comment]
This language gets less and less tiny every day.

**Comment by effectfully** [comment]
> This language gets less and less tiny every day.

Yeah, one day we'll probably compile a fair chunk of Plutus into it. And we'll probably add more "imperative" constructs.

**Comment by effectfully** [comment]
Added vectors of booleans in #34 and unpacking of integer field elements into binary form (required downstream).
